test/fuzz/yaml/yaml.go

```diff
@@ -20,10 +20,51 @@ limitations under the License.
 package yaml
 
 import (
+	"bytes"
+
 	"gopkg.in/yaml.v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	sigyaml "sigs.k8s.io/yaml"
 )
 
+// FuzzDuration is a fuzz target for unmarshaling Duration defined in "k8s.io/apimachinery/pkg/apis/meta/v1".
+// This target also checks that the unmarshaled result can be marshaled back to the input.
+func FuzzDuration(b []byte) int {
+	var unmarshalResult struct {
+		D metav1.Duration `json:"d"`
+	}
+	if err := yaml.Unmarshal(b, &unmarshalResult); err != nil {
+		return 0
+	}
+	marshalResult, err := sigyaml.Marshal(&unmarshalResult)
+	if err != nil {
+		panic(err)
+	}
+	if !bytes.Equal(marshalResult, b) {
+		panic("marshalResult != input")
+	}
+	return 1
+}
+
+// FuzzMicroTime is a fuzz target for unmarshaling MicroTime defined in "k8s.io/apimachinery/pkg/apis/meta/v1".
+// This target also checks that the unmarshaled result can be marshaled back to the input.
+func FuzzMicroTime(b []byte) int {
+	var unmarshalResult struct {
+		T metav1.MicroTime `json:"t"`
+	}
+	if err := yaml.Unmarshal(b, &unmarshalResult); err != nil {
+		return 0
+	}
+	marshalResult, err := sigyaml.Marshal(&unmarshalResult)
+	if err != nil {
+		panic(err)
+	}
+	if !bytes.Equal(marshalResult, b) {
+		panic("marshalResult != input")
```

Comment:
This would crash on inputs like:
```
t: "2019-..."
a: 10
```
since yaml.Unmarshal is not strict. This seems uninteresting. Do we want this?

Reply:
Changed to use StrictUnmarshal instead